sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/CBDE/MLIRExporter.cs

```diff
@@ -0,0 +1,853 @@
+                        case SyntaxKind.ForEachStatement:
+                        case SyntaxKind.CoalesceExpression:
+                        case SyntaxKind.ConditionalAccessExpression:
+                        case SyntaxKind.LogicalAndExpression:
+                        case SyntaxKind.LogicalOrExpression:
+                        case SyntaxKind.ForStatement:
+                        case SyntaxKind.CatchFilterClause:
+                        default:
+                            writer.WriteLine($"// Unhandled branch {bbb.BranchingNode.Kind().ToString()}");
+                            break;
+
+                    }*/
+                    break;
+                case SimpleBlock sb:
+                    writer.WriteLine($"br ^{BlockId(sb.SuccessorBlock)}");
+                    break;
+                case ExitBlock eb:
+                    // If we reach this point, it means the function has no return, we must manually add one
+                    if (HasNoReturn(parentMethod))
+                    {
+                        writer.WriteLine("return");
+                    }
+                    else
+                    {
+                        writer.WriteLine("cbde.unreachable");
+                    }
+                    break;
+
+            }
+            writer.WriteLine();
+        }
+
+        private SyntaxNode GetCondition(BinaryBranchBlock bbb)
+        {
+            // For an if or a while, bbb.BranchingNode represent the condition, not the statement that holds the condition
+            // For a for, bbb.BranchingNode represents the for. Since for is a statement, not an expression, if we
+            // see a for, we know it's at the top level of the expression tree, so it cannot be a for inside of a if condition
+            switch (bbb.BranchingNode.Kind())
+            {
+                case SyntaxKind.ForStatement:
+                    var forStmt = bbb.BranchingNode as ForStatementSyntax;
+                    return forStmt.Condition;
+                case SyntaxKind.ForEachStatement:
+                    Debug.Assert(false, "Not ready to handle those");
+                    return null;
+                default:
+                    return bbb.BranchingNode;
+            }
+        }
+
+        private string GetArgumentsString(MethodDeclarationSyntax method)
+        {
+            if (method.ParameterList.Parameters.Count == 0)
+            {
+                return string.Empty;
+            }
+            int paramCount = 0;
+            var args = method.ParameterList.Parameters.Select(
+                p => {
+                    ++paramCount;
+                    var paramName = string.IsNullOrEmpty(p.Identifier.ValueText) ?
+                        ".param" + paramCount.ToString() :
+                        EncodeName(p.Identifier.ValueText);
+                    return $"%{paramName} : {MLIRType(p)}";
+                }
+                );
+            return '(' + string.Join(", ", args) + ')';
```

Comment:
use string interpolation instead? (applies to all document)

Reply:
I usually try to use it, but in this case, I believe the concatenation is as clear (as well as being slightly faster). Maybe because I don't see this code as "a string with changing parts", but as "a string surrounded by parentheses"